### PR TITLE
[Meru AB] Instructions suggestions part 1 (UX basis)

### LINKED ISSUE #3876: [Meru AB] Suggestions for Instructions screen
Details on UX for these suggestions on  **[Figma](https://www.figma.com/file/QOxHwppG64GtPocpDhS6Y7/Product?type=design&node-id=4900-39268&mode=design&t=CrHO9TDk6FhlGRRq-4)** (also, see  context & discussion in the [making issue](https://github.com/dust-tt/decisions/issues/164))

- On no instructions, provide 3 examples of assistant instructions; static copy
- On instructions typed, call a dust app to suggest improvements, quote the issue (basically, the prompt engineering dust app)
```
pushing the user to make better instructions:
- Give examples
- What tone of voice
- What format
- For whom is the text for
- What do you want your AI to act as
- Missing
```
- 5s waiting delay (to avoid call overload);
- "Loading" from the start of the 5s delay until the suggestions are there;
- If the instructions are good, don't provide suggestions
  - Instead of red chips with suggestions, a green ship "Instructions look great :+1:"
- Ship with gating -- Dusters only for a first iteration;
- Iterate a few times to get good suggestions

### PR DESCRIPTION
**Pull request by philipperolet**
## Description
UX part for instructions suggestions #3876 with dummy suggestion generation

Todo in follow-up PR to close #3876 : 
- Plug with existing dust app `assistant-builder-instructions-suggestions`
- Confirmation pass from ed on front edge
- Prompt team to test & provide feedback

### Static suggestions
image
### Suggestions generated
image
### No suggestions (app thinks instructions are good)
image

## Risk
None (gated)

## Deploy Plan
This deploy is gated for Dust only

### CODE REVIEW

File: front/components/assistant_builder/InstructionScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+        <Collapsible.Button>
+          <div className="text-base font-bold text-element-800">
+            Suggestions
+          </div>
+        </Collapsible.Button>
+        <Collapsible.Panel>
+          <div className="flex gap-2">
+            {loading && <Spinner size="sm" />}
+            {!loading &&
+              suggestions.map((suggestion, index) => (
```
Comment (PopDaph) [inline front/components/assistant_builder/InstructionScreen.tsx:333]: NIT: Should we .slice(0, 3) to ensure we have only max 3 results?

**Review by PopDaph — APPROVED** [review]

File: front/components/assistant_builder/InstructionScreen.tsx (hunk 1 above)
Comment (philipperolet) [reply front/components/assistant_builder/InstructionScreen.tsx:333]: But I told the LLM not to output more than 3... so yeah, definitely slice :laughing:

Comment (philipperolet) [reply front/components/assistant_builder/InstructionScreen.tsx:335]: I did the fix in the following PR #3974 (in which you're pinged too, and in which the risk of having more than 3 actually materializes)

**Review by spolu — APPROVED** [review]
LGTM

Maybe we can have decent static suggestion and ship it right away?

**Comment by spolu** [comment]
We can surely come up with 3 good suggestions here or even more

**Comment by philipperolet** [comment]
@Duncid @gabhubert 
Suggestion by @spolu to ship static suggestions (and keep dynamic suggestions gated). WDYT?

=> Would require a list of 3 good suggestions. Ideally, we have 10 good suggestions and randomly show 3 of them.
If ok, could you provide copy?

**Comment by gabhubert** [comment]
Seems super random to have static suggestions no? But 

- Tip: break down your instructions into steps to leverage the model's reasoning capabilities.
- Tip: given context on how you'd like the assistant to act: "Act like a senior analyst". This leverages the underlying model's internal knowledge on the expectations of that role. 
- Tip: add instructions on the format of the answer you'd like to see: the tone of voice, whether to structure it with bullet points, in code blocks, etc...
- Tip: given precisions on the language you want the answer to be in if you'd like it to be consistent and especially if the input might be in another language
- Tip: you can ask for answers in a table and Dust will make it easy to copy. 
- Tip: Be Specific. Tailor prompts with precise language to avoid ambiguity.
- Tip: leverage examples. Provide examples within the prompt to guide the expected outcome.
- Tip: Include context. Provide relevant background information to guide responses.

**Pull request merged**